Assembling source jar (`sourcesJar` Gradle task) on generated library fails due to duplicate source directory entries

b/211697397

```
> Task :grpc-google-cloud-retail-v2alpha-java:sourcesJar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':grpc-google-cloud-retail-v2alpha-java:sourcesJar'.
> Entry com/google/cloud/retail/v2alpha/CompletionServiceGrpc.java is a duplicate but no duplicate handling strategy has been set. Please refer to https://docs.gradle.org/7.3.1/dsl/org.gradle.api.tasks.Copy.html#org.gradle.api.tasks.Copy:duplicatesStrategy for details.
```

Starting with Gradle 7, copying files fail when there are duplicate entries. Some relevant details are [here](https://github.com/gradle/gradle/issues/17236). However, unlike the workarounds suggested there, we should fix the root cause.

In our case, the cause is the following configuration
```gradle
sourceSets {
  main {
    java {
      srcDir 'src/main/java'
    }
  }
}
```
where `srcDir` [_adds_ the directory](https://docs.gradle.org/current/dsl/org.gradle.api.file.SourceDirectorySet.html#org.gradle.api.file.SourceDirectorySet:srcDir(java.lang.Object)) to the `main` source set. The [default value](https://docs.gradle.org/current/userguide/java_plugin.html#source_sets) of `main.srcDirs` is `[ 'src/main/java' ]`,  so calling `srcDir 'src/main/java'` results in `[ 'src/main/java', 'src.main/java' ]` (duplicates).

--- a/rules_java_gapic/resources/gradle/client_grpc.gradle.tmpl
+++ b/rules_java_gapic/resources/gradle/client_grpc.gradle.tmpl
@@ -33,14 +33,6 @@ test {
   exclude "**/*SmokeTest*"
 }
 
-sourceSets {
-  main {
-    java {
-      srcDir 'src/main/java'
-    }
-  }
-}
-
 clean {
   delete 'all-jars'
 }

--- a/rules_java_gapic/resources/gradle/client_grpcrest.gradle.tmpl
+++ b/rules_java_gapic/resources/gradle/client_grpcrest.gradle.tmpl
@@ -35,14 +35,6 @@ test {
   exclude "**/*SmokeTest*"
 }
 
-sourceSets {
-  main {
-    java {
-      srcDir 'src/main/java'
-    }
-  }
-}
-
 clean {
   delete 'all-jars'
 }

--- a/rules_java_gapic/resources/gradle/client_rest.gradle.tmpl
+++ b/rules_java_gapic/resources/gradle/client_rest.gradle.tmpl
@@ -30,14 +30,6 @@ test {
   exclude "**/*SmokeTest*"
 }
 
-sourceSets {
-  main {
-    java {
-      srcDir 'src/main/java'
-    }
-  }
-}
-
 clean {
   delete 'all-jars'
 }

--- a/rules_java_gapic/resources/gradle/grpc.gradle.tmpl
+++ b/rules_java_gapic/resources/gradle/grpc.gradle.tmpl
@@ -16,14 +16,6 @@ dependencies {
   {{extra_deps}}
 }
 
-sourceSets {
-  main {
-    java {
-      srcDir 'src/main/java'
-    }
-  }
-}
-
 task javadocJar(type: Jar) {
   classifier = 'javadoc'
   from javadoc

--- a/rules_java_gapic/resources/gradle/proto.gradle.tmpl
+++ b/rules_java_gapic/resources/gradle/proto.gradle.tmpl
@@ -23,9 +23,6 @@ sourceSets {
     resources {
       srcDir 'src/main/proto'
     }
-    java {
-      srcDir 'src/main/java'
-    }
   }
 }
 

